src/data/InstanceDhisRepository.ts

```diff
@@ -427,7 +426,42 @@ export class InstanceDhisRepository implements InstanceRepository {
 
     private async importEventsData(dataPackage: DataPackage): Promise<SynchronizationResult[]> {
         const events = this.buildEventsPayload(dataPackage);
-        return postEvents(this.api, events);
+
+        const programs = _(events).groupBy("program").keys().value();
```

Comment:
Let's try to always pass a function instead the string to get type checking and avoid possible typos.
`_(events).groupBy(event => event.program)` instead `_(events).groupBy("program")`